File: innlevering8/verden.py
#Klasse Verden inneholder konstruktør med fire instansvariabler, hvor to av disse ble
#initialisert med hjelpmetoder for klasse-Rutenett, en som settes lik 0 og en som 
#oppretter klasse-Rutenett ved å bruke to parametere. Det finnes totalt to metoder som
#som er ansvarlige for visning av 0 generasjon av rutenettet og det oppdateringer (overgang til neste generasjoner)
from rutenett import Rutenett

class Verden:
    def __init__(self, antall_rader, antall_kolonner):
        self._rutenett = Rutenett(antall_rader,antall_kolonner)
        self._generasjonsnummer = 0
        self._rutenett.fyll_med_tilfeldige_celler()
        self._rutenett.koble_celler()
#viser nåværende generasjon av rutenettet i terminalen til brukeren ved å kalle klasse-Rutenett
#metoden tegn_rutenett() og utskrift av ekstra output
    def tegn(self):
        self._rutenett.tegn_rutenett()
        print(f'generasjonsnummer er {self._generasjonsnummer}\nog {self._rutenett.antall_levende()} levende celler igjen')
#ved å iterere over listen 'listeNed Celle' metoden oppdaterer instansvariablen 'self._generasjonsnummer' 
#og hver celle i rutenett basert på status til naboceller 
    def oppdatering(self):
        listeMedCeller = self._rutenett.hent_alle_celler()
        for celle in listeMedCeller:
            celle.tell_levende_naboer()
        for celle in listeMedCeller:
            celle.oppdater_status()  
        self._generasjonsnummer += 1

    

    

File: Verden.java
public class Verden {
    public int antallRader;
    public int antallKolonner;
    public int genNr;
    public Rutenett rutenett;

    public Verden(int antallRader, int antallKolonner) {
        this.antallRader = antallRader;
        this.antallKolonner = antallKolonner;
        this.genNr = 0;
        this.rutenett = new Rutenett(antallRader, antallKolonner);
        this.rutenett.fyllMedTilfeldigeCeller();
        this.rutenett.kobleAlleCeller();
    }

    public void tegn () {
        this.rutenett.tegnRutenett();
        System.out.println("Generasjon nummeret er: " + this.genNr);
        System.out.println("Antall levende celler er: " + this.rutenett.antallLevende());
    }

    public void oppdatering () {
        for (int r = 0; r < this.antallRader; r++) { 
            for (int k = 0; k < this.antallKolonner; k++) { 
                Celle celle = this.rutenett.hentCelle(r, k);
                celle.tellLevendeNaboer();
            }
        }
        for (int r = 0; r < this.antallRader; r++) { 
            for (int k = 0; k < this.antallKolonner; k++) { 
                Celle celle = this.rutenett.hentCelle(r, k);
                celle.oppdaterStatus();
            }
        }
        this.genNr++;
    }
}
